lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -195,9 +201,19 @@ public ParquetReader(
                         ranges.put(new ChunkKey(columnId, rowGroup), range);
                     }
                 }
+                // Update the metrics which records the codecs used along with data size
+                long totalDataSize = ranges.get(new ChunkKey(columnId, rowGroup)).stream()
```

Comment:
Let's avoid the lookup in `ranges` and add range lengths to `totalDataSize` above just before `ranges.put`